src/codegen/build_module.cc

```diff
@@ -480,41 +500,96 @@ Array<Array<LoweredFunc> > split_dev_host_funcs(const Array<LoweredFunc>& funcs,
     fhost.Set(i, func);
   }
 
-
-  for (size_t i = 0; i < fdevice.size(); ++i) {
-    auto func = fdevice[i];
-    func = ir::LowerIntrin(func, target->target_name);
-    fdevice.Set(i, func);
-  }
-
   for (size_t i = 0; i < fhost.size(); ++i) {
     auto func = fhost[i];
-    func = ir::LowerIntrin(func, target_host_val->target_name);
+    func = ir::LowerIntrin(func, target_host->target_name);
     func = ir::CombineContextCall(func);
     fhost.Set(i, func);
   }
   return {fhost, fdevice};
 }
 
-runtime::Module build(const Array<LoweredFunc>& funcs,
-                      const Target& target,
-                      const Target& target_host,
-                      const BuildConfig& config) {
+// Create a module for a specific device (target). The lowered functions
+// associated with the host is returned as well.
+runtime::Module DeviceBuild(const Array<LoweredFunc>& funcs,
+                            const Target& target,
+                            const Target& target_host,
+                            const BuildConfig& config,
+                            Array<LoweredFunc>* fhost) {
```

Comment:
Use new split function https://github.com/dmlc/tvm/blob/master/src/codegen/build_module.cc#L426

Reply:
@antinucleon It's used below. Are you suggesting below and also here: https://github.com/dmlc/tvm/pull/3144/files#diff-8218db0c8945a45c78492fdbefb23eb8R554 to avoid passing by pointer?

Reply:
Yes. I was using pointer to get value returned and I was suggested not to do in this way.

Reply:
Done.